### PR TITLE
DO NOT MERGE: RPS fixture test to assert no regressions

### PR DESCRIPTION
**Pull request by tzaffi**
# DO NOT MERGE!!!

Adding a fixture and a unit test that sets the ground truth for how the compiler should behave. I anticipate incorporating this into #650.

### CODE REVIEW

File: tests/unit/sourcemap_precursor_test.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import json
+from pathlib import Path
+
+import pytest
+
+ALGOBANK = Path.cwd() / "examples" / "application" / "abi"
+RPS = Path.cwd() / "tests" / "teal"
```
Comment (bbroder-algo) [inline tests/unit/sourcemap_precursor_test.py:7]: nice

**Review by bbroder-algo — APPROVED** [review]

File: tests/unit/sourcemap_precursor_test.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+def compare_and_assert(file, actual):
+    with open(file, "r") as f:
+        expected_lines = f.read().splitlines()
+        actual_lines = actual.splitlines()
+        assert len(expected_lines) == len(actual_lines)
+        assert expected_lines == actual_lines
+
+
+def no_regressions_algobank():
```
Comment (tzaffi) [inline tests/unit/sourcemap_precursor_test.py:18]: moved to `tests/unit/sourcemap_test.py`

File: tests/unit/sourcemap_precursor_test.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    compare_and_assert(ALGOBANK / "algobank_clear_state.teal", clear)
+    compare_and_assert(ALGOBANK / "algobank_approval.teal", approval)
+
+
+@pytest.mark.serial
+def test_no_regression_with_sourcemap_as_configured_algobank():
+    no_regressions_algobank()
+
+
+def no_regressions_rps():
```
Comment (tzaffi) [inline tests/unit/sourcemap_precursor_test.py:38]: moved to `tests/unit/sourcemap_rps_test.py`

**Comment by tzaffi** [comment]
I'm closing now that the tests and fixtures here have been merged into #650 and it is directly mergeable against `master`.

**Pull request closed without merging**